src/main/java/rx/schedulers/ScheduleWhen.java

```diff
@@ -0,0 +1,263 @@
+import java.util.concurrent.atomic.AtomicReference;
+
+import rx.Completable;
+import rx.Completable.CompletableOnSubscribe;
+import rx.Completable.CompletableSubscriber;
+import rx.Observable;
+import rx.Scheduler;
+import rx.Subscription;
+import rx.annotations.Experimental;
+import rx.functions.Action0;
+import rx.functions.Func1;
+import rx.internal.operators.BufferUntilSubscriber;
+import rx.subjects.PublishSubject;
+import rx.subscriptions.Subscriptions;
+
+/**
+ * Allows the use of operators for controlling the timing around when actions
+ * scheduled on workers are actually done. This makes it possible to layer
+ * additional behavior on an existing {@link Scheduler}. The main parameter is a
+ * function that flattens an {@link Observable} of {@link Completable}s into
+ * just one {@link Completable}. There must be a chain of operators connecting
+ * the returned value to the source {@link Observable} otherwise any work
+ * scheduled on the {@link Scheduler} will not be executed.
+ * <p>
+ * When {@link Scheduler#createWorker()} is invoked a {@link Completable} is
+ * onNext'd to the combinator to be flattened. If the {@link Completable} is not
+ * immediately subscribed to an calls to {@link Worker#schedule} are buffered.
+ * Once the {@link Completable} is subscribed to actions are then scheduled on the
+ * actual {@link Scheduler}. When the {@link Worker} is unsubscribed the
+ * {@link Completable} emits an onComplete and triggers any behavior in the
+ * flattening operator. The {@link Observable} and all {@link Completable}s give
+ * to the flattening function never onError.
+ * <p>
+ * Slowing down the rate to no more than than 1 a second.
+ * 
+ * <pre>
+ * Scheduler slowSched = new ScheduleWhen(workers -> {
+ * 	// use concatenate to make each worker happen one at a time.
+ * 	return Completable.concat(workers.map(worker -> {
+ * 		// delay the starting of the next worker by 1 second.
+ * 		return worker.delay(1, TimeUnit.SECONDS);
+ * 	}));
+ * }, Schedulers.computation());
+ * </pre>
+ * <p>
+ * Limit the amount concurrency two at a time without creating a new fix size
+ * thread pool:
+ * 
+ * <pre>
+ * Scheduler limitSched = new ScheduleWhen(workers -> {
+ * 	// use merge max concurrent to two at a time
+ * 	return Completable.merge(workers, 2);
+ * }, Schedulers.computation());
+ * </pre>
+ */
+@Experimental
+public class ScheduleWhen extends Scheduler {
+	private final Scheduler actualScheduler;
+	private final PublishSubject<Observable<Completable>> workerQueue;
+
+	public ScheduleWhen(Func1<Observable<Observable<Completable>>, Completable> combine, Scheduler actualScheduler) {
+		this.actualScheduler = actualScheduler;
+		// workers are converted into completables and put in this queue.
+		this.workerQueue = PublishSubject.create();
+		// send it to a custom combinator to pick the order and rate at which
+		// workers are processed.
+		combine.call(workerQueue.onBackpressureBuffer()).subscribe();
```

Comment:
Do we have 'Scheduler' life cycle concerns with this one? Should this Scheduler have a close method to cancel this subscription so we can always clean up references?